firewood/tests/db.rs

```diff
@@ -57,15 +38,23 @@ fn test_basic_metrics() {
                 .max_revisions(10)
                 .build(),
         );
-    let db = Db::new("test_revisions_db2", &cfg.truncate(true).build()).unwrap();
-    let metrics = db.metrics();
+    let db = firewood::db::Db::new("test_revisions_db2", &cfg.truncate(true).build())
+        .await
+        .unwrap();
+    let metrics = db
+        .as_any()
+        .downcast_ref::<firewood::db::Db>()
+        .unwrap()
+        .metrics();
```

Comment:
If you returned the concrete type from `new`, you wouldn't have to do this. I don't see why you would want to erase the `Db` type to begin with. If we want to limit the public API, that's easy enough to do without erasing type information.